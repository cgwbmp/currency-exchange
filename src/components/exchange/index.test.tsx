import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Exchange from '.';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Exchange />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('DOM structure not changed', () => {
  const component = renderer.create(<Exchange />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
