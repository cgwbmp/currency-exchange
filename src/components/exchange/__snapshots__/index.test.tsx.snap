// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM structure not changed 1`] = `
<div
  className="ant-result ant-result-info"
>
  <div
    className="ant-result-icon"
  >
    <i
      aria-label="icon: transaction"
      className="anticon anticon-transaction"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="transaction"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M668.6 320c0-4.4-3.6-8-8-8h-54.5c-3 0-5.8 1.7-7.1 4.4l-84.7 168.8H511l-84.7-168.8a8 8 0 0 0-7.1-4.4h-55.7c-1.3 0-2.6.3-3.8 1-3.9 2.1-5.3 7-3.2 10.8l103.9 191.6h-57c-4.4 0-8 3.6-8 8v27.1c0 4.4 3.6 8 8 8h76v39h-76c-4.4 0-8 3.6-8 8v27.1c0 4.4 3.6 8 8 8h76V704c0 4.4 3.6 8 8 8h49.9c4.4 0 8-3.6 8-8v-63.5h76.3c4.4 0 8-3.6 8-8v-27.1c0-4.4-3.6-8-8-8h-76.3v-39h76.3c4.4 0 8-3.6 8-8v-27.1c0-4.4-3.6-8-8-8H564l103.7-191.6c.5-1.1.9-2.4.9-3.7zM157.9 504.2a352.7 352.7 0 0 1 103.5-242.4c32.5-32.5 70.3-58.1 112.4-75.9 43.6-18.4 89.9-27.8 137.6-27.8 47.8 0 94.1 9.3 137.6 27.8 42.1 17.8 79.9 43.4 112.4 75.9 10 10 19.3 20.5 27.9 31.4l-50 39.1a8 8 0 0 0 3 14.1l156.8 38.3c5 1.2 9.9-2.6 9.9-7.7l.8-161.5c0-6.7-7.7-10.5-12.9-6.3l-47.8 37.4C770.7 146.3 648.6 82 511.5 82 277 82 86.3 270.1 82 503.8a8 8 0 0 0 8 8.2h60c4.3 0 7.8-3.5 7.9-7.8zM934 512h-60c-4.3 0-7.9 3.5-8 7.8a352.7 352.7 0 0 1-103.5 242.4 352.57 352.57 0 0 1-112.4 75.9c-43.6 18.4-89.9 27.8-137.6 27.8s-94.1-9.3-137.6-27.8a352.57 352.57 0 0 1-112.4-75.9c-10-10-19.3-20.5-27.9-31.4l49.9-39.1a8 8 0 0 0-3-14.1l-156.8-38.3c-5-1.2-9.9 2.6-9.9 7.7l-.8 161.7c0 6.7 7.7 10.5 12.9 6.3l47.8-37.4C253.3 877.7 375.4 942 512.5 942 747 942 937.7 753.9 942 520.2a8 8 0 0 0-8-8.2z"
        />
      </svg>
    </i>
  </div>
  <div
    className="ant-result-title"
  >
    Exchange to another currency
  </div>
  <div
    className="ant-result-subtitle"
  >
    Type the amount with current currency and which currency to convert
  </div>
  <div
    className="ant-result-extra"
  >
    <div
      className="exchange"
    >
      <form
        className="ant-form ant-form-horizontal"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-control-wrapper"
            style={Object {}}
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span
                  className="ant-input-search ant-input-search-enter-button ant-input-search-large ant-input-group-wrapper ant-input-group-wrapper-lg"
                >
                  <span
                    className="ant-input-wrapper ant-input-group"
                  >
                    <input
                      aria-label="Send"
                      className="ant-input ant-input-lg"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="100 usd in euro"
                      style={null}
                      type="text"
                      value=""
                    />
                    <span
                      className="ant-input-group-addon"
                    >
                      <button
                        className="ant-btn ant-input-search-button ant-btn-primary ant-btn-lg"
                        onClick={[Function]}
                        type="button"
                      >
                        <i
                          aria-label="icon: enter"
                          className="anticon anticon-enter"
                        >
                          <svg
                            aria-hidden="true"
                            className=""
                            data-icon="enter"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M864 170h-60c-4.4 0-8 3.6-8 8v518H310v-73c0-6.7-7.8-10.5-13-6.3l-141.9 112a8 8 0 0 0 0 12.6l141.9 112c5.3 4.2 13 .4 13-6.3v-75h498c35.3 0 64-28.7 64-64V178c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </i>
                      </button>
                    </span>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
